### ISSUE TITLE
Blade::if directive with an argument of 0 is not compiled correctly

### ISSUE THREAD

**Issue by tjallingt** [laravel/framework#24390]
- Laravel Version: 5.6.23
- PHP Version: 7.2.3
- Database Driver & Version: mysql 5.7

### Description:

```php
Blade::if('iseven', function ($number) {
    return $number % 2 == 0;
});
```

Will cause errors when used with `0` (`@iseven(0)`).

At: https://github.com/laravel/framework/blob/5.6/src/Illuminate/View/Compilers/BladeCompiler.php#L394-L413

The compiler checks if the directive received any arguments and if it didn't it will output a check without an argument. However since `0` is falsy you can't pass `0` into a custom blade conditional.

### Steps To Reproduce:
Add the `iseven` blade conditional to your app service provider and add
```
@iseven(0)
test
@endiseven
```
in a blade template.

expected result would be the `test` but it will throw errors about expecting an argument that it didn't get.

**Comment by tjallingt** [comment]
@staudenmeir 
Yes, thanks for mentioning that. I still think it is kind of strange behaviour that should be fixed even if there is an easy workaround. Interestingly this doesn't seem to happen with `@iseven(false)`. I'm assuming the blade compiler is just parsing everything as text but while `"0"` is falsy `"false"` is not...

**Comment by tjallingt** [comment]
that means the fix would look like
```php
$expression || $expression === "0"
    ? /* with argument */
    :  /* without argument */;
``` 
which is kinda ugly but behaves a lot more like one would expect it to behave

**Comment by staudenmeir** [comment]
As far as I see, `$expression ? :` is meant to differentiate between statements with and without a parameter.

So I would suggest a stricter comparison: `$expression !== ''`

**Comment by tjallingt** [comment]
@staudenmeir yea that is a much cleaner way to write the same thing 👍

**Comment by staudenmeir** [comment]
Out of curiosity: Why are you using the directive with a fixed value?

**Comment by tjallingt** [comment]
Showing and hiding parts of blades depending on the "organizational level" of a user.
So although the value passed to the directive is fixed the directive compares that to a value that calculate elsewhere (so its just a convenience for writing out the whole check)

We could use names (strings) instead of numbers for the organizational levels but we tend to call them by their numbers (mostly because the names can be confusing).

**Comment by staudenmeir** [comment]
I just saw your edit: Yes, create a pull request and add a test. Since it's a bug fix, you should target the 5.6 branch.
